src/coreComponents/linearAlgebra/interfaces/hypre/HypreSolver.cpp

```diff
@@ -71,25 +72,57 @@ void HypreSolver::solve_direct( HypreMatrix & mat,
                                 HypreVector & sol,
                                 HypreVector & rhs )
 {
-  // Instantiate solver
-  HYPRE_Solver solver;
+  // Convert matrix from Matrix to SuperMatrix format
+  array1d< globalIndex > rowPtr;
+  array1d< globalIndex > cols;
+  array1d< real64 > vals;
+  SuperMatrix SLUDMat;
```

Comment:
I would put everything required by SuperLU_Dist in a `struct` as done by _hypre_:
```typedef struct
{
   HYPRE_BigInt global_num_rows;
   SuperMatrix A_dslu; 
   HYPRE_Real *berr;
   dLUstruct_t dslu_data_LU;
   SuperLUStat_t dslu_data_stat;
   superlu_dist_options_t dslu_options;
   gridinfo_t dslu_data_grid;
   dScalePermstruct_t dslu_ScalePermstruct;
   dSOLVEstruct_t dslu_solve;
}
hypre_DSLUData;
```

Reply:
This is the current structure:
https://github.com/GEOSX/GEOSX/blob/e631d57dab529aef582bc1edad51f7ac72dd42ab/src/coreComponents/linearAlgebra/interfaces/hypre/HypreSuperlu.hpp#L63-L78